src/erlcloud_sns.erl

```diff
@@ -543,6 +580,15 @@ sns_request(Config, Action, Params) ->
             erlang:error({sns_error, Reason})
     end.
 
+list_all(Fun, Type, Config, Token, Acc) ->
+    Res = Fun(Token, Config),
+    List = proplists:get_value(Type, Res),
+    case proplists:get_value(next_token, Res) of
+        "" ->
+            lists:append(Acc, List);
```

Comment:
Calling `lists:append(Acc, List)` will need to copy `Acc` each call. More efficient to use `List ++ Acc` and then reverse the result when you are done.